src/internal/i18n/provider.tsx

```diff
@@ -38,10 +38,19 @@ export function I18nProvider({ messages: messagesArray, locale: providedLocale,
   const parentMessages = useContext(I18nMessagesContext);
   const messages = mergeMessages([parentMessages, ...messagesArray]);
 
-  // If a locale isn't provided, we can try and guess from the html lang,
-  // and lastly default to English. Locales have a recommended case, but are
-  // matched case-insensitively, so we lowercase it internally.
-  const locale = (providedLocale || document?.documentElement.lang || 'en').toLowerCase();
+  let locale: string;
```

Comment:
Are you still sure you want to go your own hard way instead of reusing the existing code?
https://github.com/cloudscape-design/components/blob/main/src/internal/utils/locale/normalize-locale.ts

If not reusing, why not?